shortfin/python/shortfin/debug/array.py

```diff
@@ -0,0 +1,106 @@
+import logging
+
+import numpy as np
+
+from shortfin import array as sfnp
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+
+async def to_np(tensor: sfnp.device_array):
+    tensor_shape = tensor.shape
+
+    host_tensor = tensor.for_transfer()
+    host_tensor.copy_from(tensor)
+    await tensor.device
+    tensor = host_tensor
+
+    is_float16 = False
+    if tensor.dtype == sfnp.float16:
+        is_float16 = True
+    tensor = tensor.items
+
+    np_array = None
+
+    dtype = tensor.typecode
+    # float16 is not supported by python array, so have to
+    # explicitly set dtype for numpy.
+    if is_float16:
+        dtype = np.float16
+
+    np_array = np.frombuffer(tensor, dtype=dtype)
+
+    # Ensure shape of array matches original tensor shape
+    np_array = np_array.reshape(*tensor_shape)
+
+    return np_array
+
+
+async def dump_array(tensor: sfnp.device_array):
+    np_array = await to_np(tensor)
+    logger.debug(np_array)
+
+
+async def fill_array(tensor: sfnp.device_array, fill_value: int | float):
+    np_array = await to_np(tensor)
+    np_array.fill(fill_value)
+    return np_array
+
+
+def find_mode(arr: np.ndarray, axis=0, keepdims=False):
```

Comment:
I don't think this should be in shortfin itself. Should just be a fidget of numpy code used when needed. If it is being used all the time, can add to the public api later.

Reply:
Made `find_mode` a private method